Library/ViewModels/ProjectPageViewModelTests.swift

```diff
@@ -1140,6 +1148,194 @@ final class ProjectPageViewModelTests: TestCase {
   }
 
+  func testOutput_PresentMessageDialog() {
+    self.vm.inputs.configureWith(projectOrParam: .left(.template), refTag: nil)
+    self.vm.inputs.viewDidLoad()
+
+    self.presentMessageDialog.assertDidNotEmitValue()
+
+    self.vm.inputs.askAQuestionCellTapped()
+
+    self.presentMessageDialog.assertValues([.template])
+  }
+
+  func testOutput_ShowHelpWebViewController() {
+    self.vm.inputs.configureWith(projectOrParam: .left(.template), refTag: nil)
+    self.vm.inputs.viewDidLoad()
+
+    self.showHelpWebViewController.assertDidNotEmitValue()
+
+    self.vm.inputs
+      .projectEnvironmentalCommitmentDisclaimerCellDidTapURL(URL(string: "https://www.kickstarter.com/environment")!)
+
+    self.showHelpWebViewController.assertValues([.environment])
+  }
+
+  func testOutput_UpdateDataSourceNavigationSection() {
+    let project = Project.template
+      |> \.extendedProjectProperties .~ ExtendedProjectProperties(
+        environmentalCommitments: [],
+        faqs: [],
+        risks: "",
+        story: "",
+        minimumPledgeAmount: 1
+      )
+    let overviewSection = NavigationSection.overview.rawValue
+    let environmentalCommitmentsSection = NavigationSection.environmentalCommitments.rawValue
+
+    self.vm.inputs.configureWith(projectOrParam: .left(project), refTag: .category)
+
+    self.updateDataSourceNavigationSection.assertDidNotEmitValue()
+
+    self.vm.inputs.viewDidLoad()
+
+    self.updateDataSourceNavigationSection.assertDidNotEmitValue()
+
+    self.vm.inputs.projectNavigationSelectorViewDidSelect(index: overviewSection)
+
+    // The view model skips the first emission
+    self.updateDataSourceNavigationSection.assertDidNotEmitValue()
+
+    self.vm.inputs.projectNavigationSelectorViewDidSelect(index: environmentalCommitmentsSection)
+
+    self.updateDataSourceNavigationSection.assertValues([.environmentalCommitments])
+  }
+
+  func testOutput_UpdateDataSourceProjectProperties() {
+    let project = Project.template
+      |> \.extendedProjectProperties .~ ExtendedProjectProperties(
+        environmentalCommitments: [],
+        faqs: [],
+        risks: "",
+        story: "",
+        minimumPledgeAmount: 1
+      )
+    let overviewSection = NavigationSection.overview.rawValue
+    let environmentalCommitmentsSection = NavigationSection.environmentalCommitments.rawValue
+
+    self.vm.inputs.configureWith(projectOrParam: .left(project), refTag: .category)
+
+    self.updateDataSourceProjectProperties.assertDidNotEmitValue()
+
+    self.vm.inputs.viewDidLoad()
+
+    self.updateDataSourceProjectProperties.assertDidNotEmitValue()
+
+    self.vm.inputs.projectNavigationSelectorViewDidSelect(index: overviewSection)
+
+    // The view model skips the first emission
+    self.updateDataSourceProjectProperties.assertDidNotEmitValue()
+
+    self.vm.inputs.projectNavigationSelectorViewDidSelect(index: environmentalCommitmentsSection)
+
+    self.updateDataSourceProjectProperties.assertDidEmitValue()
+  }
```

Comment:
I'm not sure how this test and the one above are different?

Reply:
A single output which has a tuple in it was mapped into two separate observers (we have to do this because tuples are not equatable). The tests are probably identical in execution but are testing two different outputs. It looks like throughout our project we have a habit of testing these mapped observers in individual tests rather than collectively in one.